### ISSUE TITLE
A harness that vets command *shape* refuses the session-start snippet, and the skill gives no degraded form — so the protocol is skipped rather than run differently

### ISSUE THREAD

**Issue by juniordurant** [rebelytics/one-skill-to-rule-them-all#147]
**Target:** `SKILL.md` — Session Start Protocol step 2 (the scan block) and
"Id and filename" (the id snippet).
**Verified at:** upstream `main` (`7518a85`), byte-identical to v3.2.0.
**Adjacent, not duplicate:** #77 (portability beyond POSIX shells), #86/#110
(bare globs abort under zsh), #16 (GNU-only syntax on macOS), #25 (`grep -oP`
on Git-Bash). Those are about a command that RUNS and produces a wrong or
fatal result. This is about a command that is never allowed to run at all,
by a policy layer above the shell, and about what the agent is supposed to do
in the next second.

## What happened

Both snippets — the step-2 scan loop and the id computation — are multi-line
compounds: a `for` loop with an embedded `awk`, and a command substitution
wrapping a brace group, a pipe and a `sort`. Several harnesses (Claude Code's
permission layer among them, but any policy layer that classifies commands
rather than executing them) decline a compound of that shape as unverifiable:
it cannot be matched against an allowlist, so it is refused before execution.

The observed behaviour when that happens is not "run it differently". It is
one of two failures:

- the agent **rewrites the snippet into a subtler compound** — same shape,
  same refusal, two or three times — and then gives up; or
- the agent concludes the protocol is unavailable in this environment and
  proceeds with the user's task, having activated nothing.

Neither is visible afterwards. A skipped protocol leaves exactly the artefact
a protocol that found nothing leaves.

## Why the current text did not prevent it

Everything the skill says about a failing probe is framed around results —
"an empty scan in a log known to be non-empty is a broken command", "a denied
or failed write is not a read-only log", retry and try a second interface.
A *refused* command produces no result to interrogate. The nearest rule
("retry once, try a second tool that reaches the same path") is about writes,
and re-issuing the same shape is the one thing guaranteed not to help.

Nothing anywhere says: **the snippet is a convenience, the STEP is the
requirement.** So the snippet's unavailability reads as the step's
unavailability.

## Suggested fix

A short paragraph in step 2, and a matching line in "Id and filename":

> **A refused snippet is a degraded path, never a skipped step.** If the
> harness declines a command as unverifiable, do not rewrite it into a
> subtler compound — the refusal is about shape, and a cleverer compound is
> the same shape. Degrade to flat, single-purpose commands over absolute
> paths, and do the combining yourself:
>
> ```bash
> grep -H -E '^(id|title|status|skill|proposes_skill|siblings_checked):' "[ABSOLUTE PATH]/skill-observations/observation-log"/*.md
> ls "[ABSOLUTE PATH]/skill-observations/observation-log"
> ls "[ABSOLUTE PATH]/skill-observations/observation-log/archive"
> cat "[ABSOLUTE PATH]/skill-observations/observation-log/archive/.id-floor"
> ```
>
> Take the maximum of the three id sources by hand. The guard still applies:
> if the listing is non-empty and no ids were extracted, the read is broken.
> The protocol has been executed; only the mechanism changed. State in the
> session which form you used, so a later reader can tell a degraded run from
> a skipped one.

Worth stating as the general rule, because it applies to every snippet the
skill ships: an embedded command is one implementation of a step, and a
harness may refuse any of them. Every mandatory step that ships a command
needs a sentence saying what remains mandatory when the command is
unavailable.

---

Related (checked before filing, not duplicates): #77, #110, #107.

---

Reported from a production install of v3.2.0 (20 project copies, a 200-entry
log). Prepared with Claude Code (Claude Opus 5); every command quoted above was
executed verbatim from a clean `sh` before posting, and a human reviewed and
approved this report.

**Comment by moshopping** [comment]
Same failure from a second harness mode, added here instead of a duplicate issue.

**Where it happened:** an unattended session, so nobody could approve anything. Its command-approval layer refuses any command that contains a variable expansion or a command substitution unless it was approved beforehand. The step-2 block was rejected with `Contains simple_expansion`; the id block has the same shape. The session did not halt: it re-ran step 2 as a plain `ls` of the log directory, which the layer allowed. That listing reads no frontmatter, builds no status or skill awareness, and the files-versus-parsed guard never runs. This is the "skipped rather than run differently" outcome described above, with a concrete substitute.

**A complement to the degraded-path paragraph:** flat commands still leave the combining (the maximum of the three id sources, the parse guard) to be re-typed by the agent on every run, and re-typing and piping is where #100 happened. Shipping the scan and the id derivation as scripts (for example `scripts/observation-log.py scan <log-dir>` and `next-id <log-dir> <slug>`) gives a one-line call with a literal path: no expansion, no substitution, no newline, allowable once by prefix, and independent of the shell dialect. That is the shape named in #77 for commands that never run; the inline blocks can stay as the reference implementation and the degraded path.

**Related data point:** while checking this at `main` (`7518a85`), the id block also fails to parse under macOS `/bin/bash` 3.2 (#152). Same class: a mandatory step whose validity depends on the host shell.

---
Drafted with Claude Code from observations logged while using this skill; filed on the account owner's instruction.

**Comment by rebelytics** [comment]
Thank you. It's in the queue and will go into one of the next weekly releases.
